apps/ledger-live-mobile/src/screens/SyncOnboarding/LanguageSelect.tsx

```diff
@@ -0,0 +1,213 @@
+    useState<UiDrawerStatus>("none");
+
+  // Will be computed depending on the states. Updating nextDrawerToDisplay
+  // triggers the current displayed drawer to close
+  let nextDrawerToDisplay: UiDrawerStatus = "none";
+
+  const [selectedLanguage, setSelectedLanguage] = useState<string | null>(null);
+
+  const [languageSelectStatus, setLanguageSelectStatus] =
+    useState<LanguageSelectStatus>("unrequested");
+
+  // Handles a newly selected language to redux-dispatch
+  useEffect(() => {
+    if (selectedLanguage) {
+      dispatch(setLanguage(selectedLanguage));
+    }
+  }, [dispatch, selectedLanguage]);
+
+  const handleLanguageSelectOnChange = useCallback(
+    language => {
+      setSelectedLanguage(language);
+
+      if (loaded && firmwareAvailableLanguages.includes(language)) {
+        setLanguageSelectStatus("firmware-language-update-requested");
+      } else {
+        setLanguageSelectStatus("completed");
+      }
+    },
+    [firmwareAvailableLanguages, loaded],
+  );
+
+  const handleLanguageSelectOnPress = useCallback(() => {
+    setLanguageSelectStatus("language-selection-requested");
+  }, []);
+
+  const handleLanguageSelectCancel = useCallback(() => {
+    setLanguageSelectStatus("completed");
+  }, []);
+
+  const handleFirmwareLanguageUpdate = useCallback(() => {
+    // TODO: redirect to firmware localization flow when available
+  }, []);
+
+  const handleFirmwareLanguageCancel = useCallback(() => {
+    setLanguageSelectStatus("completed");
+  }, []);
+
+  // Handles the UI logic
+  if (languageSelectStatus === "language-selection-requested") {
+    nextDrawerToDisplay = "language-selection";
+  } else if (languageSelectStatus === "firmware-language-update-requested") {
+    nextDrawerToDisplay = "firmware-language-update";
+  } else {
+    nextDrawerToDisplay = "none";
+
+    // Resets entirely the drawer mechanism
+    if (currentDisplayedDrawer !== "none") {
+      setCurrentDisplayedDrawer("none");
+    }
+  }
+
+  // If there is already a displayed drawer, the currentDisplayedDrawer would be
+  // synchronized with nextDrawerToDisplay during the displayed drawer onClose event.
+  // Otherwise, currentDisplayDrawer needs to be set to nextDrawerToDisplay manually
+  if (currentDisplayedDrawer === "none" && nextDrawerToDisplay !== "none") {
+    setCurrentDisplayedDrawer(nextDrawerToDisplay);
+  }
+
+  return (
+    <Flex>
+      <Button
+        type="main"
+        outline
+        size="small"
+        Icon={ChevronBottomMedium}
+        iconPosition="right"
+        onPress={handleLanguageSelectOnPress}
+      >
+        {currentLocale.toLocaleUpperCase()}
+      </Button>
+      <BottomDrawer
+        noCloseButton
+        preventBackdropClick
+        isOpen={
+          currentDisplayedDrawer === "language-selection" &&
+          nextDrawerToDisplay === "language-selection"
+        }
+        onClose={() => setCurrentDisplayedDrawer(nextDrawerToDisplay)}
```

Comment:
possible to use a useCallback for that?

Reply:
as `BottomDrawer` is not a memoized component, it will be re-rendered each time `LanguageSelect` re-renders, even if its props did not change. So it won't change anything to memoized this function